portfolio done

BocharovTech opened the pull request.

BocharovTech committed:
portfolio done

N1SHA-git commented on js/modules/pagination.js:
пустой файл либо убрать либо перенести сюда реализацию пагинации

N1SHA-git commented on js/script.js:
```diff
@@ -0,0 +1,51 @@
+let currentPage = 1;
+
+const container = document.querySelector(".photos__gallery");
+const pagination = document.querySelector(".photos__pagination");
+
+function showPage(page) {
+  currentPage = page;
+  container.innerHTML = ""; // Очистить контейнер
+
+  const start = (page - 1) * photosPerPage;
+  const end = start + photosPerPage;
+
+  for (let i = start; i < end && i < photos.length; i++) {
+    const img = document.createElement("img");
+    img.src = photos[i];
+    img.className = "photos__img";
+    container.appendChild(img);
+  }
+
+  buildPagination();
+}
```
при пагинации происходит мерцание, попробуй в этой функции поиграться с css (display)

N1SHA-git commented on css/style.css:
```diff
@@ -0,0 +1,168 @@
+.photos__btn {
+  width: 40px;
+  height: 40px;
+  color: rgba(30, 217, 208, 0.4862745098);
```
лучше везде придерживаться либо HEX либо RGB

BocharovTech replied:
Это компилирует плагин live sass compiller, я пишу стили в scss

N1SHA-git commented on scss/blocks/_main.scss:
```diff
@@ -0,0 +1,84 @@
+      color: #888;
+      font-size: 14px;
+      align-self: flex-end;
+    }
+    &:hover {
+      transform: translateY(-5px);
+    }
```
нету transition у обычного класса

N1SHA-git commented on index.html:
```diff
@@ -0,0 +1,160 @@
+      <img src="images/logo.jpg" alt="logo" class="header__img" />
+    </header>
+    <nav class="navigation">
+      <div class="container">
```
избыточный элемент div, просто класс container в nav

N1SHA-git commented on index.html:
```diff
@@ -0,0 +1,160 @@
+    </header>
+    <nav class="navigation">
+      <div class="container">
+        <ul class="navigation__menu">
+          <li class="navigation__item">
+            <a href="#about" class="navigation__link">О себе</a>
+          </li>
+          <li class="navigation__item">
+            <a href="#hobbies" class="navigation__link">Мои хобби</a>
+          </li>
+          <li class="navigation__item">
+            <a href="#skills" class="navigation__link">Навыки</a>
+          </li>
+          <li class="navigation__item">
+            <a href="#achievements" class="navigation__link">Достижения</a>
+          </li>
+          <li class="navigation__item">
+            <a href="#contacts" class="navigation__link">Контакты</a>
+          </li>
+          <li class="navigation__item">
+            <a href="#photos" class="navigation__link">Фото</a>
+          </li>
+        </ul>
```
каждому li добавлять класс это хорошо, но можно по другому через (navigation__menu > li)

N1SHA-git commented on index.html:
```diff
@@ -0,0 +1,160 @@
+    </nav>
+
+    <main class="main">
+      <div class="container">
```
избыточен div, класс в main переноси

N1SHA-git commented on index.html:
```diff
@@ -0,0 +1,160 @@
+    <main class="main">
+      <div class="container">
+        <section id="about">
+          <h2 class="about__about">О себе</h2>
```
class about__about звучит не очень + его нету в css

N1SHA-git commented on index.html:
```diff
@@ -0,0 +1,160 @@
+        <section id="achievements">
+          <h2 class="achievements__title">Мои достижения</h2>
+
+          <article>
+            <div class="achievements__container">
+              <div class="achievements__card">
+                <h3 class="achievements__card-title">Сертификат 1</h3>
+                <p>Описание достижения или сертификата</p>
+                <span class="achievements__card-date">2023</span>
+              </div>
+              <div class="achievements__card">
+                <h3 class="achievements__card-title">Сертификат 2</h3>
+                <p>Описание достижения или сертификата</p>
+                <span class="achievements__card-date">2023</span>
+              </div>
+              <div class="achievements__card">
+                <h3 class="achievements__card-title">Сертификат 3</h3>
+                <p>Описание достижения или сертификата</p>
+                <span class="achievements__card-date">2023</span>
+              </div>
+            </div>
+          </article>
```
лучше все же через список

N1SHA-git commented on scss/blocks/_navigation.scss:
```diff
@@ -0,0 +1,23 @@
+.navigation {
+  position: sticky;
+  top: 0;
```
добавить z-index, потому что некоторые блоки залазят на navigation

N1SHA-git commented on index.html:
```diff
@@ -0,0 +1,160 @@
+          </article>
+        </section>
+
+        <section class="photos">
```
добавить id photos для якорной ссылки

N1SHA-git commented on index.html:
```diff
@@ -0,0 +1,160 @@
+          <h2 class="photos__title">Мои фотографии</h2>
+          <div class="photos__block">
+            <div class="photos__gallery"></div>
+            <div class="photos__pagination"></div>
```
добавить hover эффект для пагинации

N1SHA-git commented on css/style.css:
```diff
@@ -0,0 +1,168 @@
+  list-style: disc;
+}
+
+.skills__menu {
+  padding: 20px;
+  margin-left: 30px;
+  line-height: 150%;
```
вынести в миксин или в отдельный класс, повторяется для hobbies__menu, skills__menu, contacts__menu

yanlybimov committed:
убрал мерцание в пагинации

BocharovTech committed:
Исправил мерцание при пагинации и добавил hover эффекты на кнопки пагинации

BocharovTech committed:
Исправил ссылки на контакты.

Исправил блок с сертификатами и сделал их списком.
Немного исправил названия классов в секции achivements для улучшения логичности

BocharovTech committed:
Добавил файл с миксинами и заменил повторяющийся код

BocharovTech committed:
Merge branch 'refactoringCodeYanBocharov'
